image_writers_guide/guidelines.adoc

```diff
@@ -9,4 +9,85 @@
+
+	RUN yum -y install mypackage && yum clean all -y
+
+Note: Although the current Docker build process does not allow shrink the space used by the image when we remove something in a later layer, this may change in the future. 
+
+=== Place instructions in proper order
+Docker reads the Dockerfile and runs the instructions from top to bottom. Every successfully executed instruction creates a layer which can be reused next we build this (or other image). It is very important to place on top of your Dockerfile instructions that will change rarely.
+
+This will make sure the next builds of the same image will be very fast because the cache will not be invalidated by upper layer changes.
+
+=== Use EXPOSE to mark important ports
+See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `EXPOSE` Important Ports_
+
+=== Set environment variables with ENV
+It’s a good idea to set environment variables with the `ENV` instruction. One example is to set version of your project. This will make it easy for people find the version without even looking at the Dockerfile.  Another example of an environment variable worth setting is when to advertise some path on the system that could be used by some process, like `JAVA_HOME`.
+
+=== Do not set default passwords
+It is not a good idea to set default passwords. Many people will extend the image and forget to remove or change the default password. This will lead to security issues where in production there will be a user with a well-known passwords. Passwords should be configurable via environment variable instead (See anchor:configuration[])
+
+=== Do not start (or install) SSHD
+Running SSHD should be avoided.  For accessing running containers you can use `docker exec` locally, or the OpenShift tooling which allows execution of arbitrary commands in running images.  Installing and running SSHD in your image opens up additional vectors for attack and requirements for security patching.
+
+== OpenShift Specific Recommendations
+=== Enable images for Source-To-Image
+https://github.com/openshift/source-to-image[Source to Image] (STI) is a build tool which makes it easy to write images that take application source code as an input, and produce a new image that will run the assembled application as output.  For example, this https://github.com/openshift/wildfly-8-centos[Wildfly] image defines STI scripts which will run a `maven` build on a java source repository and copy the resulting war file into the Wildfly deployments directory.  The resulting image will now automatically start Wildfly with the application running.
+
+For more details about how to write STI scripts for your image, see the https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md[STI documentation].
+
+=== Use Volumes for Persisted Data
```

Comment:
I would move this to general use.  Make anything that generates temporary data that might live longer than a container in a volume.  Write true temporary data to /tmp, and clean up after yourself.

Also, Docker 1.5 adds a "readonly" container flag that might be used in production environments to encourage image authors to use volumes.